### PR TITLE
[#148] Add coredns

### LINKED ISSUE #148: Add coredns service
We need some dns server (https://github.com/nspcc-dev/coredns) to resolve bucket names to container ids in s3-gw.

### PR DESCRIPTION
**Pull request by KirillovDenis**
This PR adds new service: `coredns`. New coredns image is expected to be uploaded to dockerhub or existed locally.
To build local image use:
```
$ git clone https://github.com/coredns/coredns
$ cd coredns
$ make
$ docker build --rm -f Dockerfile -t nspcssdev/coredns:v016 .
```` 

Should be merged and considered together with https://github.com/nspcc-dev/neofs-s3-gw/pull/284 (since not release version of s3 is used)

closes #148 

Signed-off-by: Denis Kirillov <denis@nspcc.ru>

### CODE REVIEW

File: services/coredns/Corefile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+. {
+    nns http://192.168.130.90:30333
+    transfer {
+       to *
+    }
+    log
+    debug
+    forward . 8.8.8.8:53
```
Comment (realloc) [inline services/coredns/Corefile:8]: Local docker DNS service should be used to avoid confusion

File: services/coredns/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    stop_signal: SIGKILL
+    volumes:
+      - ./Corefile:/Corefile
+      - ./../../vendor/hosts:/etc/hosts
+
+networks:
+  chain_int:
+  internet:
+    external: true
+    name: basenet_internet
```
Comment (realloc) [inline services/coredns/docker-compose.yml:23]: \n

File: services/s3_gate/docker-compose.yml [hunk 3]
```diff
@@ -8,6 +8,8 @@ services:
     hostname: s3_gate
     container_name: s3_gate
     restart: on-failure
+    dns: 
+      - ${IPV4_PREFIX}.53
```
Comment (realloc) [inline services/s3_gate/docker-compose.yml:12]: :+1:

**Review by realloc — APPROVED** [review]
LGTM, didn't test.

File: .env [hunk 4]
```diff
@@ -23,9 +23,13 @@ HTTP_GW_VERSION=0.16.0
 HTTP_GW_IMAGE=nspccdev/neofs-http-gw
 
 # S3 Gate
-S3_GW_VERSION=0.17.0
+S3_GW_VERSION=0.17.0-15-gda9f272
```
Comment (alexvanin) [inline .env:26]: Debug?

Comment (KirillovDenis) [reply .env:26]: I've thought we need to update s3-gw  image (by new release or intermediate image) to be able to check coredns works together with s3-gw correctly. Though we can check coredns separately (making dns request to `coredns` container).

Comment (alexvanin) [reply .env:26]: I don't mind to bump version of S3 gateway, we use intermediate version in dev-env so it is okay, but it should be available on DockerHub.

**Review by alexvanin — APPROVED** [review]

**Pull request merged**